specification/resources/resource-manager/Microsoft.Authorization/stable/2023-01-01/privateLinks.json

```diff
@@ -0,0 +1,409 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "PrivateLinksClient",
+    "version": "2023-01-01",
+    "description": "To manage and control access to your private link resources"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.Management/managementGroups/{groupId}/providers/Microsoft.Authorization/privateLinkAssociations/{plaId}": {
+      "put": {
+        "tags": [
+          "PrivateLinkAssociation"
+        ],
+        "operationId": "PrivateLinkAssociation_Put",
+        "description": "Create a PrivateLinkAssociation",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/GroupIdParameter"
+          },
+          {
+            "$ref": "#/parameters/PrivateLinkAssociationIdParameter"
+          },
```

Comment:
I want to add a parameter which will be passed as a request header. Checking if you have done that before?

Reply:
Do you have an example of how that would look in sample request?

Reply:
Header: Auxiliary Token
Header Name :: x-ms-authorization-auxiliary
Header Value :: Bearer , EncryptedBearer , Bearer

This is what we want to add in the request header

Reply:
Thinking of getting it from parameter during PUT.

"parameters": {
    "plaId": "00000000-0000-0000-0000-000000000000",
    "groupId": "my-management-group",
    "api-version": "2023-01-01",
    "AuxiliaryToken": "Bearer xxxxxx"
    "parameters": {
      "privateLink": "00000000-0000-0000-0000-000000000000",
      "publicNetworkAccess": "Enabled",
      "tenantId": "06b3ccb8-1384-4bcc-aec7-852f6d57161b",
      "linkedPrivateLinkTenantId": "06b3ccb8-1384-4bcc-aec7-852f6d570000"
    }
  },

Reply:
In the PUT example, that would made the AuxilaryToken a path variable. To make a parameter you would include in parameters request body